beacon-chain/core/blocks/block_operations.go

```diff
@@ -727,56 +686,118 @@ func ProcessValidatorDeposits(
 	}
 
 	valIndexMap := stateutils.ValidatorIndexMap(beaconState)
-	for idx, deposit := range deposits {
-		if err = verifyDeposit(beaconState, deposit); err != nil {
-			return nil, fmt.Errorf("could not verify deposit #%d: %v", idx, err)
+	for _, deposit := range deposits {
+		beaconState, err = ProcessDeposit(beaconState, deposit, valIndexMap, verifySignatures, true)
+		if err != nil {
+			return nil, fmt.Errorf("could not process deposit index %d: %v", deposit.Index, err)
 		}
-		beaconState.DepositIndex++
-		pubKey := deposit.Data.Pubkey
-		amount := deposit.Data.Amount
-		index, ok := valIndexMap[bytesutil.ToBytes32(pubKey)]
-		if !ok {
-			if verifySignatures {
-				// TODO(#2307): Use BLS verification of proof of possession.
-			}
-			effectiveBalance := amount - (amount % params.BeaconConfig().EffectiveBalanceIncrement)
-			if params.BeaconConfig().MaxEffectiveBalance < effectiveBalance {
-				effectiveBalance = params.BeaconConfig().MaxEffectiveBalance
-			}
-			beaconState.ValidatorRegistry = append(beaconState.ValidatorRegistry, &pb.Validator{
-				Pubkey:                     pubKey,
-				WithdrawalCredentials:      deposit.Data.WithdrawalCredentials,
-				ActivationEligibilityEpoch: params.BeaconConfig().FarFutureEpoch,
-				ActivationEpoch:            params.BeaconConfig().FarFutureEpoch,
-				ExitEpoch:                  params.BeaconConfig().FarFutureEpoch,
-				WithdrawableEpoch:          params.BeaconConfig().FarFutureEpoch,
-				EffectiveBalance:           effectiveBalance,
-			})
-			beaconState.Balances = append(beaconState.Balances, amount)
-		} else {
-			beaconState = helpers.IncreaseBalance(beaconState, uint64(index), amount)
+	}
+	return beaconState, nil
+}
+
+// ProcessDeposit takes in a deposit object and inserts it
+// into the registry as a new validator or balance change.
+//
+// Spec pseudocode definition:
```

Comment:
godoc format seems off, please fix this : )

<img width="663" alt="Screen Shot 2019-06-13 at 7 25 46 AM" src="https://user-images.githubusercontent.com/21316537/59440934-9aa66c00-8dac-11e9-9a0c-4fdf50b9c0fc.png">